### DISCUSSION TITLE
Rename "renounce" to "disableCancelability"

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#777] [Ideas]
The task is to rename the `renounce` functionality (the function name, the event, etc.) to `disableCancelability`.

The rationale is two-fold:

1. Since the introduction of the `transferable` flag in V2.1, there are now multiple properties of the stream that could be "renounced" by the sender.
2. "disable cancelability" is infinitely clearer than "renounce". The latter could be interpreted to mean that the sender gives up on paying the recipient.

**Comment by razgraf** [comment]
A note that might slightly influence these refactoring (not this one specifically):

Any renaming of methods or events increases the complexity of the app and subgraphs due to backwards compatibility requirements. In the case of new flags, it's understandable why the change in signature is necessary, but for slight improvements in nomenclature, it might simply be worth setting the change aside (even though in this case, the suggested change makes the action more intuitive or contextual).

These increases in complexity influence not only our own internal development time (duplicating event handling in subgraphs, conditionally calling a function based on contract version) but also our integrators' time supporting these updates.

P.S. Not saying we shouldn't rename the concept in the OP, just added an argument worth taking into account when discussing similar matters.

**Reply by PaulRBerg** [reply]
Makes sense.

At a minimum, we will always have the option to implement the changes only at the front-end level, i..e display the button as "Disable Cancelability" but keep calling the `renounce` function in the smart contract.

**Comment by PaulRBerg** [comment]
Gonna close this discussion since it's a very low priority.
